toolchain/check/convert.cpp

```diff
@@ -1139,7 +1139,10 @@ auto Convert(Context& context, SemIR::LocId loc_id, SemIR::InstId expr_id,
   }
 
   // We can only perform initialization for complete, non-abstract types.
-  if (!RequireConcreteType(
+  // Facet types are okay though, since their representation is fixed and we
+  // want to support using the `Self` of an interface inside its definition.
+  if (!context.types().Is<SemIR::FacetType>(target.type_id) &&
+      !RequireConcreteType(
```

Comment:
This special case makes me a bit uncomfortable. If there's some set of types we can initialize / convert to while they're not concrete, I'd like to have a name for that, and a `RequireBlah` function for that, rather than a special case here.

I wonder if that name *is* "concrete", and we should say that facet types are always concrete even when they're not complete.

Reply:
Changed `RequireConcreteType` to return true for facet types. It didn't seem to affect tests.